### ISSUE TITLE
Related issues on test wer.

### ISSUE THREAD

**Issue by mengjing556** [ms-dot-k/Lip-to-Speech-Synthesis-in-the-Wild#5]
Hi,author!How many epochs did you train, or do you have any other settings? I loaded the visual front-end you provided and trained for 50 epochs with the same settings as you, but I can't get the same result on wer as your paper said. and the other audio evaulations are correct?
<img width="638" alt="1701395040441" src="https://github.com/ms-dot-k/Lip-to-Speech-Synthesis-in-the-Wild/assets/65903955/a259b23b-dfb2-442a-9756-499b4de03458">
<img width="227" alt="1701394961167" src="https://github.com/ms-dot-k/Lip-to-Speech-Synthesis-in-the-Wild/assets/65903955/87a0c103-76d3-4f13-830a-32576e45aec7">

**Comment by ms-dot-k** [comment]
We trained the model about 80 epochs.
How did you measure the WER?
Can you compare the WERs with the pre-trained model I uploaded?

**Comment by mengjing556** [comment]
wer = total editdistance of the batch / total tokens(truth text) of the batch.
Yes, I loaded the pretrain_weights of visual frontend.
Maybe I can continue to train for about 30 epochs? 
Thank you for your reply, I will try to find if somewhere is wrong.

**Comment by ms-dot-k** [comment]
I mean there is no pre-trained ASR in this repo to measure the WER. So I’m wondering which model did you use.

Also I uploaded pretrained lip to speech model. so can you measure the WER of the pretrained system with the ASR model you used?


-----Original Message-----
보낸사람: ***@***.***>
받는사람: ***@***.***>
참조: "Minsu ***@***.***>, ***@***.***>
날짜: 2023. 12. 4 오후 8:14:36
제목: Re: [ms-dot-k/Lip-to-Speech-Synthesis-in-the-Wild] Related issues on test wer. (Issue #5)

wer = total editdistance of the batch / total tokens(truth text) of the batch.
Yes, I loaded the pretrain_weights of visual frontend.
Maybe I can continue to train for about 30 epochs?
Thank you for you reply, I will try to find if somewhere is wrong.

**Comment by mengjing556** [comment]
I used the pre-trained asr model(actually mel to text) that you provided in the train.py,and I measured the WER of the pretrained system as shown in the figure:
<img width="568" alt="1701693714786" src="https://github.com/ms-dot-k/Lip-to-Speech-Synthesis-in-the-Wild/assets/65903955/2df63a2a-9cfe-4a07-b84f-3dc6240423ee">
I misunderstood, I thought the WER given in the paper was obtained through the pretrained asr model(mel to text) your provided.

**Comment by mengjing556** [comment]
I found the problem and fixed it. I made an error in the mask settings of ASR. I have solved this problem now. Thank you, author!
<img width="230" alt="1701715550427" src="https://github.com/ms-dot-k/Lip-to-Speech-Synthesis-in-the-Wild/assets/65903955/48c1e59a-ac96-4ecc-9dd7-0f7975b22cb2">
